modules/primo/includes/primo.search.inc

```diff
@@ -178,7 +178,23 @@ function primo_search_search(TingSearchRequest $ting_query) {
 
   // Handle raw query.
   if (!empty($raw_query = $ting_query->getRawQuery())) {
-    $queries['query'][] = htmlspecialchars($raw_query);
+    // The query starts with "[", so we query for comma separated ID's.
+    if (strpos($raw_query, '[') !== FALSE) {
+      $raw_ids = explode(',', trim($raw_query, '[]'));
```

Comment:
Why not just straight ```$ids[] = explode(',', trim($raw_query, '[]'));``` ? (notice i added a space to the trim chars btw.
Also consider doing an $ids = array_map('trim', $ids) afterwards to clear out any whitespace around the ids